# HG changeset patch
# User John W. Eaton <jwe@octave.org>
# Date 1232728105 18000
#      Fri Jan 23 11:28:25 2009 -0500
# Node ID 6adcafc70c321cc2299a98535d86840099be79e8
# Parent  188d38a553c73e5b99e3d1b7c338a6661e548c87
toplev.cc (octave_call_stack::do_goto_caller_frame): allow caller frame to be base frame

diff --git a/src/ChangeLog b/src/ChangeLog
--- a/src/ChangeLog
+++ b/src/ChangeLog
@@ -1,8 +1,13 @@
+2009-01-23  John W. Eaton  <jwe@octave.org>
+
+	* toplev.cc (octave_call_stack::do_goto_caller_frame):
+	Also allow caller frame to be base frame, not just user code.
+
 2009-01-23  Jaroslav Hajek  <highegg@gmail.com>
 
 	* gripes.cc (gripe_indexed_cs_list, gripe_invalid_inquiry_subscript):
 	New functions.
 	* gripes.h: Declare them.
 	* pt-idx.cc: Remove definitions of the above funcs.
 	* ov-cell.cc (octave_cell::subsref): Declare constants as const.
 	(octave_cell::subsasgn): Remove dead branch, declare constants as const.
diff --git a/src/toplev.cc b/src/toplev.cc
--- a/src/toplev.cc
+++ b/src/toplev.cc
@@ -432,17 +432,17 @@ octave_call_stack::do_goto_caller_frame 
   while (frame != 0)
     {
       frame = cs[frame].prev;
 
       const call_stack_elt& elt = cs[frame];
 
       octave_function *f = elt.fcn;
 
-      if (f && f->is_user_code ())
+      if (frame == 0 || (f && f->is_user_code ()))
 	{
 	  if (! skipped)
 	    // We found the current user code frame, so skip it.
 	    skipped = true;
 	  else
 	    {
 	      // We found the caller user code frame.
 	      call_stack_elt tmp (elt);
